src/table/components/TableTotals.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { memo, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import TableRow from '@mui/material/TableRow';
+import TableCell from '@mui/material/TableCell';
+import { useContextSelector, TableContext } from '../context';
+import { getHeaderStyle, getTotalsCellStyle } from '../utils/styling-utils';
+import { totalHandleKeyPress } from '../utils/handle-key-press';
+import { handleClickToFocusCell } from '../utils/handle-accessibility';
+
+function TableTotals({ rootElement, tableData, theme, layout, translator, keyboard }) {
+  const { columns, paginationNeeded, totalsPosition, rows } = tableData;
+  const setFocusedCellCoord = useContextSelector(TableContext, (value) => value.setFocusedCellCoord);
+  const totalHeaderStyle = useMemo(() => getHeaderStyle(layout, theme), [layout, theme.name()]);
+  const headRowStyle = {
+    '& :last-child': {
+      borderRight: paginationNeeded && 0,
+    },
+    'th:first-of-type': {
+      borderLeft: !paginationNeeded && '1px solid rgb(217, 217, 217)',
+    },
+  };
```

Comment:
fyi I have merged a change that changes how we style things, so this wont be needed when you merge main :)